### PR TITLE
Add success return data to UPE blocks payment processing

### LINKED ISSUE #2649: [GlobalStep] "Missing value for stripe.confirmSetup():" error message is displayed for checkout block when UPE is enabled.
## Bug Description:
When UPE is enabled and user tries to complete the checkout from page with a Checkout block using new payment method, "Missing value for stripe.confirmSetup():" error is displayed.

## Environment:
Woocommerce Version : WooCommerce 5.5.2
WooCommerce Payments - 2.8.0-rc3.

PC:
Windows 10, Mac 10.14.6
Chrome(Version 91.0.4472.77) 
Firefox(Version 88.0)
Safari: v14.1.2

## Steps To Reproduce:
1. Create any test site using JN site.
1. Install and activate all the required plugins.
1. Upload and activate the WooCommerce Payments v2.8.0-rc3 plugin. 
1. Complete the setup wizard.
1. Complete the KYC flow.
1. Install and activate WooCommerce Blocks plugin.
1. Create a page with Cart and Checkout blocks.
1. As a shopper add a product to the cart.
1. Visit to page with Checkout block.
1. Click on "Use another payment method" option.
1. Enter a valid card details and click on "Place Order" button.
1. Observe that "Missing value for stripe.confirmSetup():" error message is displayed.

## Note:
Order is placed successfully at backend even after getting the error.

## Actual Result:
"Missing value for stripe.confirmSetup():" error message is displayed for checkout block when UPE is enabled.

## Expected Result:
User should be able to place order from checkout block without any error when UPE is enabled.

## Screenshot:

*Isolating the problem (mark completed items with an [x]):*
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

<details>

`
### WordPress Environment ###

WC Version: 5.5.2
REST API Version: ✔ 5.5.2
WC Blocks Version: ✔ 5.6.0
Action Scheduler Version: ✔ 3.2.1
WC Admin Version: ✔ 2.4.4
Log Directory Writable: ✔
WP Version: 5.8
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.48 (Unix) OpenSSL/1.0.2g
PHP Version: 7.4.21
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.33-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.5.2
WC Database Prefix: wp_
Total Database Size: 5.44MB
Database Data Size: 3.82MB
Database Index Size: 1.62MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.08MB + Index: 0.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.06MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ms_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 2.48MB + Index: 0.08MB + Engine InnoDB
wp_postmeta: Data: 0.23MB + Index: 0.20MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 24
page: 8
post: 2
product: 22
product_variation: 7
revision: 3
shipping_package: 2
shop_order: 28
shop_order_refund: 3
shop_subscription: 4

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (8) ###

Query Monitor: by John Blackbourn – 3.7.1
Code Snippets: by Code Snippets Pro – 2.14.1
Companion Plugin: by Osk – 1.18
WooCommerce Blocks: by Automattic – 5.6.0
WooCommerce Advanced Shipping Packages: by Jeroen Sormani – 1.1.8
WooCommerce Payments: by Automattic – 2.8.0-rc
WooCommerce Subscriptions: by WooCommerce – 3.1.3
WooCommerce: by Automattic – 5.5.2

### Inactive Plugins (6) ###

Akismet Anti-Spam: by Automattic – 4.1.10
Google Listings and Ads: by WooCommerce – 1.3.0
Hello Dolly: by Matt Mullenweg – 1.7.2
Jetpack: by Automattic – 9.9.1
Mailchimp for WooCommerce: by Mailchimp – 2.5.1
MailPoet 3 (New): by MailPoet – 3.65.1

### Dropin Plugins (1) ###

db.php: Query Monitor Database Class

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.7.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: https://sensible-snow.jurassic.ninja
Subscription Statuses: wc-active: 4
WooCommerce Account Connected: ❌ No
Report Cache Enabled: ✔ Yes
Cache Update Failures: ✔ 0 failure

### Store Setup ###

Country / State: United States (US) — California

### Subscriptions by Payment Gateway ###

WooCommerce Payments: wc-active: 4

### Payment Gateway Support ###

WooCommerce Payments: products
refunds
subscriptions
subscription_cancellation
subscription_suspension
subscription_reactivation
subscription_amount_changes
subscription_date_changes
subscription_payment_method_change
subscription_payment_method_change_customer
subscription_payment_method_change_admin
multiple_subscriptions
tokenization
add_payment_method

### Action Scheduler ###

Complete: 164
Oldest: 2021-08-03 10:59:32 +0000
Newest: 2021-08-03 15:24:05 +0000

Failed: 1
Oldest: –
Newest: –

Pending: 8
Oldest: 2021-08-04 13:25:31 +0000
Newest: 2021-11-01 11:51:41 +0000

### Status report information ###

Generated at: 2021-08-03 15:24:08 +00:00
`

</details>

### PR DESCRIPTION
**Pull request by bborman22**
Fixes #2649 

#### Changes proposed in this Pull Request

The goal of this PR is to fix an issue with the blocks checkout using UPE. The error identified in the issue was caused by the UPE blocks checkout sending a saved payment method to process rather than the newly entered card data. This has been addressed in this PR by explicitly returning a success message with the payment method data. Previously the non-explicit response caused unexpected behavior and led to the error encountered.

#### Testing instructions

* Start with a site using WooCommerce Blocks and UPE enabled.
* The customer must have at least one saved payment method on their account.
* Go to checkout with a product using the block based checkout with the customer with a saved payment method.
* At checkout select to use a new payment method and provide the relevant payment details.
* Confirm that the order is processed successfully and that the order was charged to the newly created payment method.

-------------------

- [ ] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

**Review by FangedParakeet — APPROVED** [review]
This change makes sense and mirrors the expected `onPaymentProcessing` response that is reflected in the regular gateway equivalent file for the Blocks Checkout. I tested with WC 5.5.2 and Blocks 5.6 and was happily able to checkout with a new payment method using the Blocks checkout, with the appropriate payment information being attached to the resultant order. 

Nice one figuring out the fix so rapidly. LGTM.

**Review by RadoslavGeorgiev — APPROVED** [review]
Tested with a slightly older Blocks version (5.5.0) and now the issue is gone. The code makes sense and I don't think it could cause any regressions either.

TL;DR: LGTM, 🚢 it :)
